cuda_bindings/tests/test_cufile.py

```diff
@@ -1886,3 +1905,632 @@ def test_set_get_parameter_string():
+
+    err, ctx = cuda.cuDevicePrimaryCtxRetain(device)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+    (err,) = cuda.cuCtxSetCurrent(ctx)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+
+    # Open cuFile driver
+    cufile.driver_open()
+
+    try:
+        # Set statistics level first (required before starting stats)
+        cufile.set_stats_level(1)  # Level 1 = basic statistics
+
+        # Start collecting cuFile statistics
+        cufile.stats_start()
+
+        # Verify statistics collection is active
+        # Note: Additional verification would require stats_get() or similar functions
+        logging.info("cuFile statistics collection started successfully")
+
+    finally:
+        # Close cuFile driver
+        cufile.driver_close()
+        cuda.cuDevicePrimaryCtxRelease(device)
+
+
+@pytest.mark.skipif(
+    cufileVersionLessThan(1150), reason="cuFile parameter APIs require cuFile library version 13.0 or later"
+)
+def test_stats_stop():
+    """Test cuFile statistics collection stop."""
+    # Initialize CUDA
+    (err,) = cuda.cuInit(0)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+
+    err, device = cuda.cuDeviceGet(0)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+
+    err, ctx = cuda.cuDevicePrimaryCtxRetain(device)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+    (err,) = cuda.cuCtxSetCurrent(ctx)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+
+    # Open cuFile driver
+    cufile.driver_open()
+
+    try:
+        # Set statistics level first (required before starting stats)
+        cufile.set_stats_level(1)  # Level 1 = basic statistics
+        # Start collecting cuFile statistics first
+        cufile.stats_start()
+
+        # Stop collecting cuFile statistics
+        cufile.stats_stop()
+
+        # Verify statistics collection is stopped
+        logging.info("cuFile statistics collection stopped successfully")
+
+    finally:
+        # Close cuFile driver
+        cufile.driver_close()
+        cuda.cuDevicePrimaryCtxRelease(device)
+
+
+@pytest.mark.skipif(
+    cufileVersionLessThan(1150), reason="cuFile parameter APIs require cuFile library version 13.0 or later"
+)
+def test_stats_reset():
+    """Test cuFile statistics reset."""
+    # Initialize CUDA
+    (err,) = cuda.cuInit(0)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+
+    err, device = cuda.cuDeviceGet(0)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+
+    err, ctx = cuda.cuDevicePrimaryCtxRetain(device)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+    (err,) = cuda.cuCtxSetCurrent(ctx)
+    assert err == cuda.CUresult.CUDA_SUCCESS
+
+    # Open cuFile driver
+    cufile.driver_open()
+
+    try:
+        # Set statistics level first (required before starting stats)
+        cufile.set_stats_level(1)  # Level 1 = basic statistics
+        # Start collecting cuFile statistics first
+
+        cufile.stats_start()
+
+        # Reset cuFile statistics to clear all counters
+        cufile.stats_reset()
```

Comment:
Q: Does this call also reset the stats level? If not, we should do it in `finally`

Reply:
ditto